src/common/dav/dav_iface.c

```diff
@@ -57,42 +77,58 @@ setup_dav_phdr(dav_obj_t *hdl)
 static void
 persist_dav_phdr(dav_obj_t *hdl)
 {
-	pmemops_persist(&hdl->p_ops, hdl->do_phdr, offsetof(struct dav_phdr, dp_unused));
+	mo_wal_persist(&hdl->p_ops, hdl->do_phdr, offsetof(struct dav_phdr, dp_unused));
 }
 
 static dav_obj_t *
-dav_obj_open_internal(int fd, int flags, size_t sz)
+dav_obj_open_internal(int fd, int flags, size_t sz, const char *path, struct umem_store *store)
 {
-	int rc;
 	dav_obj_t *hdl = NULL;
 	void *base;
 	char *heap_base;
 	uint64_t heap_size;
 	int persist_hdr = 0;
+	int err = 0;
+	int rc;
 
 	base = mmap(NULL, sz, PROT_READ|PROT_WRITE, MAP_SHARED, fd, 0);
 	if (base == MAP_FAILED) {
 		close(fd);
 		return NULL;
 	}
 
-	D_ALIGNED_ALLOC(hdl, 64, sizeof(dav_obj_t));
+	D_ALIGNED_ALLOC(hdl, CACHELINE_SIZE, sizeof(dav_obj_t));
 	if (hdl == NULL) {
-		errno = ENOMEM;
+		err = ENOMEM;
 		goto out1;
 	}
 
+	/* REVISIT: In future pass the meta instance as argument instead of fd */
 	hdl->do_fd = fd;
 	hdl->do_base = base;
 	hdl->do_size = sz;
+	hdl->p_ops.base = hdl;
```

Comment:
It will be cleaned up in the future. Right now there are interfaces that depend on the p_ops as argument which inturn tries to get the dav handle back.  On pmdk the handle was stored in the base address. In dav the handle is stored outside of persistent heap.